cg/apps/demultiplex/sample_sheet/sample_sheet_creator.py

```diff
@@ -136,3 +112,108 @@ def construct_sample_sheet(self) -> List[List[str]]:
         )
         LOG.info("Sample sheet passed validation")
         return sample_sheet_content
+
+
+class SampleSheetCreatorV1(SampleSheetCreator):
+    """Create a raw sample sheet for NovaSeq6000 flow cells."""
+
+    def add_dummy_samples(self) -> None:
+        """Add all dummy samples with non-existing indexes to samples.
+
+        Dummy samples are added if there are indexes that are not used by the actual samples.
+        This means that we will add each dummy sample (that is needed) to each lane.
+        """
+        LOG.info("Adding dummy samples for unused indexes")
+        indexes_by_lane: Dict[int, Set[str]] = index.get_indexes_by_lane(samples=self.lims_samples)
+        for lane, lane_indexes in indexes_by_lane.items():
+            LOG.debug(f"Add dummy samples for lane {lane}")
+            for index_obj in self.valid_indexes:
+                if index.index_exists(index=index_obj.sequence, indexes=lane_indexes):
+                    LOG.debug(f"Index {index_obj.sequence} already in use")
+                    continue
+                dummy_flow_cell_sample: FlowCellSample = get_dummy_sample(
+                    flow_cell_id=self.flow_cell_id,
+                    dummy_index=index_obj.sequence,
+                    lane=lane,
+                    name=index_obj.name,
+                    sample_type=self.sample_type,
+                )
+                LOG.debug(f"Adding dummy sample {dummy_flow_cell_sample} to lane {lane}")
+                self.lims_samples.append(dummy_flow_cell_sample)
+
+    def get_additional_sections_sample_sheet(self) -> List[List[str]]:
+        """Build all sections of the sample sheet that is not the data section."""
+        return [
+            [SampleSheetNovaSeq6000Sections.Settings.HEADER.value],
+            SampleSheetNovaSeq6000Sections.Settings.BARCODE_MISMATCH_INDEX1.value,
+            SampleSheetNovaSeq6000Sections.Settings.BARCODE_MISMATCH_INDEX2.value,
+        ]
+
+    def get_data_section_header_and_columns(self) -> List[List[str]]:
+        """Return the header and column names of the data section of the sample sheet."""
+        return [
+            [SampleSheetNovaSeq6000Sections.Data.HEADER.value],
+            SampleSheetNovaSeq6000Sections.Data.COLUMN_NAMES.value[self.bcl_converter],
+        ]
+
+
+class SampleSheetCreatorV2(SampleSheetCreator):
+    """Create a raw sample sheet for NovaSeqX flow cells."""
+
+    def __init__(
+        self,
+        bcl_converter: str,
+        flow_cell: FlowCellDirectoryData,
+        lims_samples: List[FlowCellSample],
+        force: bool = False,
+    ):
+        super().__init__(bcl_converter, flow_cell, lims_samples, force)
+        if bcl_converter == BclConverter.BCL2FASTQ:
+            raise SampleSheetError(f"Can't use {BclConverter.BCL2FASTQ} with sample sheet v2")
+        self.bcl_converter: str = BclConverter.DRAGEN
```

Comment:
As far as I am aware, there is the BCL_CONVERT, DRAGEN and BCL2FASTQ options here, although support might only exist for the two latter ones. Is this line intended to make sure that BCL_CONVERT and DRAGEN result in the same logic being executed? I thought that BCL_CONVERT need not be run on dragen.

Reply:
Fair enough, if there is no intention to run BclConvert off-dragen in the future, it would have no effect. But if there would be some other enum which would not work with sample sheet v2 it would probably be easier to have "if bcl_converter != BclConverter.DRAGEN".